src/librustc_codegen_llvm/abi.rs

```diff
@@ -396,6 +403,8 @@ impl<'tcx> FnAbiLlvmExt<'tcx> for FnAbi<'tcx, Ty<'tcx>> {
             llvm::Attribute::NoReturn.apply_llfn(llvm::AttributePlace::Function, llfn);
         }
 
+        unwind(llfn, self.can_unwind);
```

Comment:
You don't need a separate function, IMO it's more readable if it looks like the `noreturn` code.

Reply:
You can also add a comment like `NoReturn` has.